pages/articles/SectionsHandler.inc.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * @file pages/articles/SectionsHandler.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SectionsHandler
+ * @ingroup pages_articles
+ *
+ * @brief Handle requests for sections functions.
+ *
+ */
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\handler\Handler;
+use APP\security\authorization\OjsJournalMustPublishPolicy;
+use APP\submission\Submission;
+use APP\template\TemplateManager;
+use PKP\db\DAORegistry;
+use PKP\security\authorization\ContextRequiredPolicy;
+
+class SectionsHandler extends Handler
```

Comment:
I'd like to see as much of this code moved to pkp-lib as possible. The only difference between sections (OJS) and series (OMP) is the database table names and a few other names in the system. We will eventually consolidate them so that "sections" and "series" are only in the user-facing descriptions.

Something like this in pkp-lib:

```php
<?php
namespace PKP\pages\publication;

use APP\facades\Repo;
use APP\handler\Handler;
use APP\submission\Collector;
use Illuminate\Support\LazyCollection;
use PKP\context\Context;
use PKP\context\PKPSection;

class SectionHandler extends Handler
{
    protected string $table = 'sections';
    protected string $settingsTable = 'section_settings';
    
    public function authorize(...)
    {
        //
    }

    public function section(...)
    {
        // ...

        $collector = $this->getCollector($section->getId(), $context->getId());

        $total = Repo::submission()->getCount($collector);

        $submissions = Repo::submission()->getCount(
            $collector
                ->limit($limit)
                ->offset($offset)
        );

        // ...

        $this->assignTemplateVars($submissions, $section, $context);

        $templateMgr->display('frontend/pages/section.tpl');
    }

    abstract protected function getCollector(int $sectionId, int $contextId): Collector;
    abstract protected function assignTemplateVars(LazyCollection $submissions, PKPSection $section, Context $context);
}
```

And then in OJS you add the required methods:

```php
<?php
namespace APP\pages\publication;

use APP\core\Application;
use APP\facades\Repo;
use APP\submission\Collector;
use APP\submission\Submission;
use APP\template\TemplateManager;
use Illuminate\Support\LazyCollection;
use PKP\context\Context;
use PKP\context\PKPSection;

class SectionHandler extends \PKP\pages\publication\SectionHandler
{
    protected function getCollector(int $sectionId, int $contextId): Collector
    {
        return Repo::submission()
            ->getCollector()
            ->filterByContextIds([$contextId])
            ->filterBySectionIds([$sectionId])
            ->filterByStatus(...)
            ->orderBy(...);
    }

    protected function assignTemplateVars(LazyCollection $submissions, PKPSection $section, Context $context)
    {
        $issues = Repo::issue()->getMany(
            Repo::issue()
                ->getCollector()
                ->filterByContextIds([$context->getId()])
                ->filterBySubmissionIds([$submissions->map(fn(Submission $submission) => $submission->getId())])
        );

        $templateMgr = TemplateManager::getManager(Application::get()->getRequest());
        $templateMgr->assign([
            //
        ]);
    }
}
```

Reply:
I anticipated this and already wrote OJS and OPS sectionhandlers very close to eachother.  Thanks for the examples!

Reply:
So just checking, that are you suggesting we do not use the old naming convention where we would have `PKPSectionHandler` in pkp-lib?